src/main/java/com/o19s/es/ltr/feature/store/StoredFeatureNormalizers.java

```diff
@@ -0,0 +1,197 @@
+        builder.startObject(); // begin feature norms
+        for (Map.Entry<String, FeatureNormDefinition> ftrNormDefEntry: featureNormalizers.entrySet()) {
+            builder.field(ftrNormDefEntry.getKey());
+            ftrNormDefEntry.getValue().toXContent(builder, params);
+        }
+        builder.endObject(); // feature norms
+        return builder;
+    }
+
+    public void addFeatureNormalizers(List<FeatureNormDefinition> ftrNormDefs) {
+        this.featureNormalizers = new HashMap<>();
+        for (FeatureNormDefinition ftrNorm: ftrNormDefs) {
+            this.featureNormalizers.put(ftrNorm.featureName(), ftrNorm);
+        }
+    }
+
+    public FeatureNormalizerSet compile(FeatureSet featureSet) {
+        if (featureNormalizers.size() == 0) {
+            return new NoOpFeatureNormalizerSet();
+        }
+        List<Normalizer> ftrNorms = new ArrayList<>(featureSet.size());
+
+        for (int i = 0; i < featureSet.size(); i++) {
+            ftrNorms.add(new NoOpNormalizer());
+        }
+
+        for (Map.Entry<String, FeatureNormDefinition> ftrNormDefEntry: featureNormalizers.entrySet()) {
+            String featureName = ftrNormDefEntry.getValue().featureName();
+            Normalizer ftrNorm = ftrNormDefEntry.getValue().createFeatureNorm();
+
+            if (!featureSet.hasFeature(featureName)) {
+                throw new ElasticsearchException("Feature " + featureName +
+                                                 " not found in feature set " + featureSet.name());
+            }
+
+            int ord = featureSet.featureOrdinal(featureName);
+            ftrNorms.set(ord, ftrNorm);
+        }
+        return new CompiledFeatureNormalizerSet(ftrNorms);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof StoredFeatureNormalizers)) return false;
+        StoredFeatureNormalizers that = (StoredFeatureNormalizers) o;
+
+        return that.featureNormalizers.equals(this.featureNormalizers);
+    }
+
+    @Override
+    public int hashCode() {
+        return this.featureNormalizers.hashCode();
+    }
+
+
+    private  FeatureNormDefinition createFromStreamInput(StreamInput input) throws IOException {
+        Type normType = Type.readFromStream(input);
+        if (normType == Type.STANDARD) {
+            return new StandardFeatureNormDefinition(input);
+        } else if (normType == Type.MIN_MAX) {
+            return new MinMaxFeatureNormDefinition(input);
+        }
+        // note the Type constructor throws on this condition as well
+        throw new ElasticsearchException("unknown normalizer type during deserialization");
+    }
+
+    public void writeTo(StreamOutput output) throws IOException {
+        output.writeInt(this.featureNormalizers.size());
+        for (Map.Entry<String, FeatureNormDefinition> featureNormEntry: this.featureNormalizers.entrySet()) {
```

Comment:
same here if you have the implementation declared in the registry simply calling `output.writeNamedWriteableList(new ArrayList(this.featureNormalizers.values()))`